ddtrace/contrib/trace_utils.py

```diff
@@ -372,12 +372,15 @@ def int_service(pin, int_config, default=None):
         return cast(str, int_config.service_name)
 
     global_service = int_config.global_config._get_service()
-    if global_service:
+    if global_service and global_service != int_config.global_config._inferred_service:
```

Comment:
This seems weird, are we expecting or seeing the integration config (`int_config`) to have a global config with an inferred service that isn't the global config's (`config`) service?

Reply:
I put a comment explaining why. TLDR is `global_service = config.service`, and if no `DD_SERVICE` is set, then config.service defaults to our inferred base service. However, for v0, we would still have `global_service=None` for this function, so we would be breaking a ton of customer service names without this check.

Reply:
ah, ok, this is a wierd redirection but its also used elsewhere.

Reply:
Yeah it is weird logic to add, but necessary to not break customers. I also added the same logic to the `_get_service` function as well.